server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/model/Keyword.scala

```diff
@@ -69,21 +52,23 @@ object Keyword {
       | must not contain characters with hex from '\\u0000' to '\\u00019'
       | or {'(' ')' '{' ']' '%' '*' '\"' '\\'} """.stripMargin
 
-  def parse(flagName: String): Option[Keyword] = {
-    if (isValid(flagName)) return Some(new Keyword(flagName))
-    throw new IllegalArgumentException(VALIDATION_MESSAGE)
+  def parse(flagName: String): Either[String, Keyword] = {
+    if (isValid(flagName)) Right(new Keyword(flagName))
+    Left(VALIDATION_MESSAGE)
   }
 
-  def of(flagName: String): Keyword = parse(flagName).get
+  def of(flagName: String): Keyword = parse(flagName) match {
```

Comment:
IMO you should return a Try instead of throwing

Reply:
Not done ?